File: WebApi/Interfaces/IErpService.ts
import type { CariDto, DepoDto, OnHandQuantityDto, OpenGoodsForOrderByCustomerDto, OpenGoodsForOrderDetailDto, ProjeDto, StokDto } from '../Models/index';
import type { ApiResponse, PagedResponse } from '../ApiResponse';

export interface IErpService {
  getOnHandQuantityById(depoKodu: number, stokKodu: string, seriNo: string, projeKodu: string): Promise<ApiResponse<OnHandQuantityDto>>;
  getCaris(): Promise<ApiResponse<CariDto[]>>;
  getStoks(): Promise<ApiResponse<StokDto[]>>;
  getDepos(): Promise<ApiResponse<DepoDto[]>>;
  getProjeler(): Promise<ApiResponse<ProjeDto[]>>;
  getOpenGoodsForOrderByCustomerById(cariKodu: string): Promise<ApiResponse<OpenGoodsForOrderByCustomerDto>>;
  getOpenGoodsForOrderDetailsByOrderNumbers(orderNumber: string): Promise<ApiResponse<OpenGoodsForOrderDetailDto[]>>;
}

File: Interfaces/IErpService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface IErpService
    {
        // OnHandQuantity işlemleri
          Task<ApiResponse<OnHandQuantityDto?>> GetOnHandQuantityByIdAsync(int? depoKodu = null, string? stokKodu = null, string? seriNo = null, string? projeKodu = null);

        // Cari işlemleri
        Task<ApiResponse<List<CariDto>>> GetCarisAsync();

        // Stok işlemleri
        Task<ApiResponse<List<StokDto>>> GetStoksAsync();

        // Depo işlemleri
        Task<ApiResponse<List<DepoDto>>> GetDeposAsync();

        // Proje işlemleri
        Task<ApiResponse<List<ProjeDto>>> GetProjelerAsync();

        // Müşteri siparişi işlemleri
        Task<ApiResponse<OpenGoodsForOrderByCustomerDto?>> GetOpenGoodsForOrderByCustomerByIdAsync(string cariKodu);

        // Sipariş detay işlemleri
        Task<ApiResponse<List<OpenGoodsForOrderDetailDto>>> GetOpenGoodsForOrderDetailsByOrderNumbersAsync(string orderNumber);
    }
}